### DISCUSSION TITLE
Support comments in CSV/TSV rendering

### DISCUSSION THREAD

**Discussion by rajeee** [community/community#44616] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Product Feedback

### Body

Currently, the TSV/CSV rendering engine doesn't support comments. However, [it is common to have comments in CSV files](https://github.com/NREL/resstock/blob/develop/project_national/housing_characteristics/Ceiling%20Fan.tsv), and popular tools, such as pandas [read_csv](https://pandas.pydata.org/docs/reference/api/pandas.read_csv.html) support ignoring comments. It would be nice if github.com could also support ignoring comments.

**Comment by elbosso** [comment]
There are however many different formats for comments out there - as is illustrated for example [here](https://github.com/mcnees/Solar-Neighborhood/blob/master/StarsCartesian.csv)

**Reply by ashundi-tibco** [reply]
So, either pick a comment character (`#`, `%`, etc) or allow user to configure it. Both are acceptable and better than the status quo.

**Reply by ghost** [reply]
assume all lines with incorrect number of tabs are comments. "correct" is decided by whatever number of tab chars is most common across all rows of the files.

if the number of tabs is correct, assume it is not a comment.

dont use comment chars like #, %, to decide comments IMO.

**Reply by VladimirAlexiev** [reply]
@samueldashadrach that's very unsafe and will prevent finding errors in CSVs

**Reply by VladimirAlexiev** [reply]
@elbosso
- the W3C CSVW spec has [CSV Dialect](https://w3c.github.io/csvw/metadata/#dialect-descriptions) that includes a [comment-prefix](https://w3c.github.io/csvw/syntax/#dfn-comment-prefix) that by default is `#`.
- Are there other CSV specs that allow comment lines?
- (https://datatracker.ietf.org/doc/html/rfc4180 doesn't allow comment lines)
